#! /bin/sh /usr/share/dpatch/dpatch-run
## 01.boost_limits.dpatch by Piotr Roszatycki <dexter@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Add support for other Debian platforms.

@DPATCH@
diff -urNad lyx-1.3.6/boost/boost/detail/limits.hpp /tmp/dpep.xmEzuo/lyx-1.3.6/boost/boost/detail/limits.hpp
--- lyx-1.3.6/boost/boost/detail/limits.hpp	2005-07-14 18:31:05.000000000 +1000
+++ /tmp/dpep.xmEzuo/lyx-1.3.6/boost/boost/detail/limits.hpp	2005-09-13 16:00:49.385385296 +1000
@@ -54,8 +54,18 @@
 #elif defined(__i386__) || defined(__alpha__) || defined(__ia64) || defined(__ia64__)
 #define BOOST_LITTLE_ENDIAN
 #else
+
+// Added for Debian GNU/Linux systems not handled above.
+#include <endian.h>
+#if __BYTE_ORDER == __BIG_ENDIAN
+#define BOOST_BIG_ENDIAN
+#elif __BYTE_ORDER == __LITTLE_ENDIAN
+#define BOOST_LITTLE_ENDIAN
+#else
+
 #error The file boost/detail/limits.hpp needs to be set up for your CPU type.
 #endif
+#endif
 
 namespace std {
 
